api/task_test.go

```diff
@@ -0,0 +1,204 @@
+package api
+
+import (
+	"errors"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+
+	"github.com/hashicorp/consul-terraform-sync/config"
+	"github.com/hashicorp/consul-terraform-sync/driver"
+	"github.com/hashicorp/consul-terraform-sync/event"
+	mocks "github.com/hashicorp/consul-terraform-sync/mocks/driver"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+)
+
+func TestTask_New(t *testing.T) {
+	cases := []struct {
+		name    string
+		version string
+	}{
+		{
+			"happy path",
+			"v1",
+		},
+	}
+
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+			h := newTaskHandler(event.NewStore(), map[string]driver.Driver{}, tc.version)
+			assert.Equal(t, tc.version, h.version)
+		})
+	}
+}
+
+func TestTask_ServeHTTP(t *testing.T) {
+	t.Parallel()
+	cases := []struct {
+		name       string
+		path       string
+		method     string
+		body       string
+		statusCode int
+	}{
+		{
+			"patch update task",
+			"/v1/tasks/task_patch_update",
+			http.MethodPatch,
+			`{"enabled": true}`,
+			http.StatusOK,
+		},
+		{
+			"put update task",
+			"/v1/tasks/xyz",
+			http.MethodPut,
+			``,
+			http.StatusMethodNotAllowed,
+		},
+	}
+
+	drivers := make(map[string]driver.Driver)
+	patchUpdateD := new(mocks.Driver)
+	patchUpdateD.On("UpdateTask", mock.Anything).Return(nil).Once()
```

Comment:
I'm seeing a lot of `mock.Anything` in mocked method signatures throughout and just wanted to callout that this would skip out on testing that the function is receiving the correct parameter. It's useful for tests that aren't testing that specific parameter.

Not looking for any change here, just wanted to share more about the `mock` testing options. You could optionally make the test more granular to specify `UpdateTaskConfig{Enabled: true}` for the test "patch update task".